File: day2/1.py
with open("input.txt") as f:
    inp = f.readlines()

inp = list(map(lambda x: x.split(), inp))

def solve(line):
    
    min_count, max_count = line[0].split('-')
    min_count, max_count = int(min_count), int(max_count)

    char = line[1][0]

    return min_count <= line[2].count(char) <= max_count

if __name__=="__main__":
    print(sum(list(map(solve, inp))))

File: day2/1.cc
#include<iostream>
#include<string>
#include<fstream>
#include<vector>
#include<sstream>
#include<algorithm>

int main () {

    std::ifstream inp("input.txt");
    std::string line;
    
    int out = 0;

    while(std::getline(inp,line)) {

        std::stringstream ss(line);
        std::string tmp;
        std::vector<std::string> words;

        while(std::getline(ss,tmp,' ')) {
            words.push_back(tmp);
        }

        size_t split_point = words[0].find("-");

        int min = std::stoi(words[0].substr(0,split_point));
        int max = std::stoi(words[0].substr(split_point+1));
    
        std::string password;
        char ch;
        ch = words[1][0];
        password = words[2];

        size_t count = std::count(password.begin(), password.end(), ch);

        if((count <= max) && (count >= min)) out += 1;

    }

    std::cout << out << std::endl;
    return 0;

}